src/table/genericPreviewTable.jsx

```diff
@@ -97,17 +119,16 @@ class GenericPreviewTable extends Component {
     this.state.sortableColumnPropery === columnProps.column.property,
   );
 
-  handleSelected = ({ id }) => this.setState((prevState) => {
+  handleSelected = row => this.setState((prevState) => {
+    let currentRow;
     const rows = prevState.rows.map((item) => {
-      if (item.id !== id) {
+      if (item[this.props.rowKey] !== row[this.props.rowKey]) {
```

Comment:
This condition looks kinda off, maybe consider flipping the operand and write it like this
```javascript
const rows = prevState.rows.map((item) => {
  if (item[this.props.rowKey] === row[this.props.rowKey]) {
    item.selected = !item.selected;
    currentRow = {...item};
  }
  return item;
})
```